From: Yifeng <wuyifeng@nd.com.cn>
Date: Mon, 25 Apr 2022 07:00:00 +0000
Subject: AP9-5181 Hide Chromium Bookmark functionality and Settings

Fix version: 1.1.0
Hide the bookmark from the toolbar and the menu.
Hide the Settings from the menu.

---
 chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java | 5 +++++
 chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarTablet.java | 4 +++-
 2 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java b/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/app/appmenu/AppMenuPropertiesDelegateImpl.java
@@ -554,6 +554,11 @@ public class AppMenuPropertiesDelegateImpl implements AppMenuPropertiesDelegate
         // Only display the Enter VR button if VR Shell Dev environment is enabled.
         menu.findItem(R.id.enter_vr_id).setVisible(isCurrentTabNotNull && shouldShowEnterVr());

+        // AP9-5181 Hide Bookmark functionality and Settings
+        menu.findItem(R.id.all_bookmarks_menu_id).setVisible(false);
+        menu.findItem(R.id.bookmark_all_tabs_menu_id).setVisible(false);
+        menu.findItem(R.id.preferences_id).setVisible(false);
+
         updateManagedByMenuItem(menu, currentTab);
     }

diff --git a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarTablet.java b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarTablet.java
--- a/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarTablet.java
+++ b/chrome/browser/ui/android/omnibox/java/src/org/chromium/chrome/browser/omnibox/LocationBarTablet.java
@@ -58,6 +58,7 @@ class LocationBarTablet extends LocationBarLayout {
         mBookmarkButton = findViewById(R.id.bookmark_button);
         mSaveOfflineButton = findViewById(R.id.save_offline_button);

+        mBookmarkButton.setVisibility(View.GONE);
         boolean isRtl = mUrlActionContainer.getLayoutDirection() == LAYOUT_DIRECTION_RTL;
         int urlActionContainerPadding =
                 getResources().getDimensionPixelSize(R.dimen.location_bar_url_action_padding);
@@ -232,7 +233,8 @@ class LocationBarTablet extends LocationBarLayout {
     }

     /* package */ void setBookmarkButtonVisibility(boolean showBookmarkButton) {
-        mBookmarkButton.setVisibility(showBookmarkButton ? View.VISIBLE : View.GONE);
+        // From AP9-5181, we will hide the bookmark button.
+        // so this method won't do anything.
     }

     /* package */ void setSaveOfflineButtonVisibility(
--
2.17.1